[draft] Case studies (#46416)
mui/material-ui: docs/pages/customers/tebra.md

image: '/static/branding/companies/tebra_spotlight.svg'
date: 2022-06-08T00:00:00.000Z
tags: ['MUI X']
rank: '2'
manualCard: true
---

<style>
  #blog-responsive-image {
    height: 230px;
    @media (max-width: 600px) {
      height: 167px;
    }
  }
</style>

# Tebra | Modernizing legacy reporting systems in healthcare

<img
    id="blog-responsive-image"
    src="/static/branding/companies/headers/tebra-header.png"
    alt="Tebra logo"
    style="width: 100%; height: auto; object-fit: cover; object-position: top left; border: 0px; margin-left: 0; margin-bottom: 20px; display: block; text-align: left;"
  />

## Overview

[Tebra](https://tebra.com/) is a healthcare technology company that has been in the market for over a decade.
As their platform evolved over the years, they accumulated a mix of legacy technologies including JSPs, Angular, and various React implementations, creating inconsistent user experiences across their platform.

<span class="only-light-mode">
<blockquote style="margin: 32px 0; padding: 24px 32px; background: #f5f5f7; border-left: 6px solid var(--muidocs-palette-primary-main); border-radius: 8px; font-size: 1.15rem; font-style: italic;">

>   The feature of searching in the grid is absolutely a killer for us. Everything loads to the client, the data sits in memory inside the grid, and now you can search anything—and the search is so fast and amazing.

Reviewer: ```suggestion
  Searching in the grid is sn absolute killer feature for us. Everything loads to the client, the data sits in memory inside the grid, and now you can search anything—and the search is so fast and amazing.
```

This might be seen as a negative. The data grid also supports server-side search.

- Inconsistent user experience - Users encountered dramatically different experiences depending on which part of the platform they accessed - some screens felt "old and antiquated" while others offered modern functionality.
- Performance constraints - The team had to implement complex throttling and pagination systems to manage large datasets, creating architectural constraints that limited reporting capabilities.

## Solution

Here's how Tebra adopted the MUI X Data Grid as the cornerstone of their modernization strategy and solved the aforementioned challenges:

- Unified design system - They built a comprehensive design system using MUI components, with the Data Grid serving as the foundation for all reporting experiences across the platform.

- Advanced grid capabilities - The team implemented sophisticated features including:
  - Dynamic filtering with custom external filter components
  - Fast client-side search across up to 10,000 rows
  - Grouping and summarization functionality
  - Customizable row density for improved readability
  - Advanced sorting capabilities

> - Architectural optimization - They moved from database-heavy pagination to client-side data loading, enabling lightning-fast search and filtering while maintaining performance through strategic data limits.

Reviewer: No lazy-loading?
